coral-common/src/main/java/com/linkedin/coral/common/transformers/StandardUDFOperatorTransformer.java

```diff
@@ -0,0 +1,260 @@
+
+  // Operators allowed in the transformation
+  static {
+    OP_MAP.put("+", SqlStdOperatorTable.PLUS);
+    OP_MAP.put("-", SqlStdOperatorTable.MINUS);
+    OP_MAP.put("*", SqlStdOperatorTable.MULTIPLY);
+    OP_MAP.put("/", SqlStdOperatorTable.DIVIDE);
+    OP_MAP.put("^", SqlStdOperatorTable.POWER);
+    OP_MAP.put("%", SqlStdOperatorTable.MOD);
+    OP_MAP.put("date", new SqlUserDefinedFunction(new SqlIdentifier("date", SqlParserPos.ZERO), ReturnTypes.DATE, null,
+        OperandTypes.STRING, null, null));
+    OP_MAP.put("timestamp", new SqlUserDefinedFunction(new SqlIdentifier("timestamp", SqlParserPos.ZERO),
+        FunctionReturnTypes.TIMESTAMP, null, OperandTypes.STRING, null, null) {
+      @Override
+      public void unparse(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+        // for timestamp operator, we need to construct `CAST(x AS TIMESTAMP)`
+        Preconditions.checkState(call.operandCount() == 1);
+        final SqlWriter.Frame frame = writer.startFunCall("CAST");
+        call.operand(0).unparse(writer, 0, 0);
+        writer.sep("AS");
+        writer.literal("TIMESTAMP");
+        writer.endFunCall(frame);
+      }
+    });
+    OP_MAP.put("hive_pattern_to_trino",
+        new SqlUserDefinedFunction(new SqlIdentifier("hive_pattern_to_trino", SqlParserPos.ZERO),
+            FunctionReturnTypes.STRING, null, OperandTypes.STRING, null, null));
+  }
+
+  public static final String OPERATOR = "op";
+  public static final String OPERANDS = "operands";
+  /**
+   * For input node:
+   * - input equals 0 refers to the result
+   * - input great than 0 refers to the index of source operand (starting from 1)
+   */
+  public static final String INPUT = "input";
+  public static final String VALUE = "value";
+  public static final String REGEX = "regex";
+  public static final String NAME = "name";
+
+  public final String fromOperatorName;
+  public final int numOperands;
+  public final SqlOperator targetOperator;
+  public List<JsonObject> operandTransformers;
+  public JsonObject resultTransformer;
+  public List<JsonObject> operatorTransformers;
+
+  public StandardUDFOperatorTransformer(@Nonnull String fromOperatorName, int numOperands,
+      @Nonnull SqlOperator targetOperator, @Nullable String operandTransformers, @Nullable String resultTransformer,
+      @Nullable String operatorTransformers) {
+    this.fromOperatorName = fromOperatorName;
+    this.numOperands = numOperands;
+    this.targetOperator = targetOperator;
+    if (operandTransformers != null) {
+      this.operandTransformers = parseJsonObjectsFromString(operandTransformers);
+    }
+    if (resultTransformer != null) {
+      this.resultTransformer = new JsonParser().parse(resultTransformer).getAsJsonObject();
+    }
+    if (operatorTransformers != null) {
+      this.operatorTransformers = parseJsonObjectsFromString(operatorTransformers);
+    }
+  }
+
+  @Override
+  protected boolean predicate(SqlCall sqlCall) {
```

Comment:
I realize this was called `condition` originally. Thinking again, I think `condition` is more accurate since `predicate` does not need to be satisfied (yes referring to changing the API method name).